### PR TITLE
Fix/335 guard classifier random weights

### PR DESCRIPTION
**Pull request by MaheshReddy-ML**
Guard Classifier Random Weights Bug Fix Summary #335 
==============================================

Issue
-----
The maintainer reported that the Guard module loaded microsoft/deberta-v3-small
directly when no fine-tuned classifier checkpoint was present. Because the code
used AutoModelForSequenceClassification with num_labels=3 on the base DeBERTa
model, HuggingFace created a new classification head with random weights.

That made prompt injection classification unreliable. The Guard API could return
inconsistent allow, sanitize, or block decisions for the same style of prompt,
especially when the regex layer did not fully determine the outcome.

What I Changed
---------------
1. Removed the unsafe runtime fallback to a random classification head.

   The backend classifier and standalone SDK classifier now only load the
   transformer model when a real trained checkpoint exists on disk.

   A valid checkpoint can contain either:
   - pytorch_model.bin
   - model.safetensors

   If no trained checkpoint is found, the runtime no longer creates a base
   DeBERTa sequence-classification model with random head weights.

2. Added a deterministic heuristic fallback.

   When a fine-tuned model is unavailable, the classifier now uses stable,
   deterministic rules based on the existing regex filter plus extra common
   prompt-injection patterns.

   This fallback classifies prompts as:
   - malicious for high-risk injection or jailbreak attempts
   - suspicious for lower-risk prompt disclosure or manipulation language
   - benign for normal prompts

   This is not a replacement for a fine-tuned semantic classifier, but it is
   much safer than returning predictions from random model weights.

3. Preserved fine-tuned model support.

   If a trained model is placed in the configured classifier directory, the
   Guard still loads and uses the transformer classifier as intended.

   The training path still supports loading the base DeBERTa model explicitly
   when training is requested. The change only prevents random base-model
   inference during normal Guard runtime.

4. Updated model detection.

   The backend config and SDK model detection now recognize both common
   HuggingFace weight formats:
   - pytorch_model.bin
   - model.safetensors

5. Added regression tests.

   New tests verify that when the fine-tuned model is missing:
   - the classifier uses the deterministic fallback
   - the random DeBERTa head path is not called
   - known injection prompts produce stable malicious/suspicious results
   - benign prompts remain benign

6. Updated documentation.

   The docs previously stated that using the base DeBERTa model with random
   classification head weights was the default behavior. That was the source of
   the bug, so the docs now explain the corrected behavior:

   - trained checkpoint available: use fine-tuned DeBERTa classifier
   - trained checkpoint unavailable: use deterministic heuristic fallback
   - fine-tuning is still recommended for stronger semantic coverage

Files Updated
-------------
Backend:
- backend/app/modules/guard/intent_classifier.py
- backend/app/modules/guard/guard_config.py
- backend/app/modules/guard/llm_guard.py
- backend/tests/test_guard.py

Standalone Guard SDK:
- guard-sdk/src/aegisai_guard/intent_classifier.py
- guard-sdk/src/aegisai_guard/llm_guard.py
- guard-sdk/tests/test_guard_sdk.py
- guard-sdk/README.md

Documentation:
- docs/guard-module.md
- docs/getting-started.md
- docs/architecture.md

Why This Fix Solves the Problem
-------------------------------
The original bug happened because the system made predictions using a classifier
head that had never been trained. Random weights are not meaningful for security
decisions, so the Guard could behave unpredictably.

After this fix, Guard decisions are no longer based on random model outputs.
Runtime behavior is now deterministic when no trained model is installed, and
the system still supports loading a proper fine-tuned classifier whenever one is
provided.

This means a fresh install of AegisAI will not silently rely on random ML
predictions for prompt injection detection.

Verification
------------
A syntax-level verification was completed successfully with:

python3 -m py_compile backend/app/modules/guard/intent_classifier.py backend/app/modules/guard/guard_config.py guard-sdk/src/aegisai_guard/intent_classifier.py backend/tests/test_guard.py guard-sdk/tests/test_guard_sdk.py

Result
------
The reported issue is addressed:

- The Guard classifier no longer uses random classification head weights during
  normal runtime.
- Missing fine-tuned models now trigger deterministic fallback logic.
- Documentation now matches the corrected behavior.
- Regression tests were added to prevent the random-head fallback from returning.

And I would request you to label this as GSSoC Approved

### CODE REVIEW

**Review by SdSarthak — COMMENTED** [review]
This fixes a real safety issue — deterministic heuristics when no fine-tuned model is present is far better than random DeBERTa weights silently producing garbage classifications. The lazy-import __init__.py also resolves the module-level torch import problem correctly.

The PR has a merge conflict — likely due to recent changes in the guard module. Please rebase onto main and this should be ready.

**Comment by MaheshReddy-ML** [comment]
@SdSarthak Can you please label this as **GSSoC Approved**
